tools/godotcpp.py

```diff
@@ -529,8 +528,22 @@ def generate(env):
 
 
 def _godot_cpp(env):
-    extension_dir = normalize_path(env.get("gdextension_dir", env.Dir("gdextension").abspath), env)
-    api_file = normalize_path(env.get("custom_api_file", env.File(extension_dir + "/extension_api.json").abspath), env)
+    extension_dir = env.get("gdextension_dir", None)
+    # A user provided extension_dir is relative to where they are running from (#)
+    # But our gdextension directory is here in the source tree
+    if extension_dir is not None:
+        extension_dir = normalize_path(extension_dir, env)
+    else:
+        extension_dir = env.Dir("gdextension").srcnode().abspath
+
+    api_file = env.get("custom_api_file", None)
+    # A user provided api_file is relative to where they are running from (#)
+    # But a default api_file is relative to extension_dir
+    if api_file is not None:
```

Comment:
This is the same logic twice, should probably abstract this to make it more readable.

Reply:
I used the `SCons` `converter=` feature for the variables; makes more sense anyway because it's applied to the variable across all `env` uses, not just when retrieving it.